### PR TITLE
Improve symbolic shape analysis for divisibility (and fix bug)

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135825

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -1878,6 +1878,7 @@ def solve(self):
         # NOTE(avik): We do not need to solve them for symbols that have already been specialized.
         reduced_congruences = self._reduce_congruences()
         for s, congruences in reduced_congruences.items():
+            # break
```
Comment (ezyang) [inline torch/fx/experimental/symbolic_shapes.py:1881]: remove

File: torch/fx/experimental/symbolic_shapes.py [hunk 2]
```diff
@@ -4382,9 +4383,11 @@ def get_axioms(self, symbols: Optional[Tuple["sympy.Symbol"]] = None, compute_hi
                                for rs in self.deferred_runtime_asserts.values()
                                for r in rs)
         else:
-            runtime_asserts = (r.expr
+            runtime_asserts = [r.expr
                                for s in symbols if s not in self.var_to_val
-                               for r in self.deferred_runtime_asserts.get(s, ()))
+                               for r in self.deferred_runtime_asserts.get(s, ())]
+            runtime_asserts += [r.expr
+                               for r in self.deferred_runtime_asserts.get(None, ())]
```
Comment (ezyang) [inline torch/fx/experimental/symbolic_shapes.py:4390]: sigh

**Review by ezyang — APPROVED** [review]
test plz

**Comment by ZainRizvi** [comment]
Please rebase onto a more recent commit. This older build depends on runner types that no longer exist

**Pull request closed without merging**